engine/texc/src/texc_buffer.cpp

```diff
@@ -59,61 +59,45 @@ namespace dmTexc
         dmZlib::DeflateBuffer(data, size, 9, &out_data_array, DeflateWriter);
         size = out_data_array.Size();
 
-        uint8_t* out_data = (uint8_t*)malloc(size);
-        memcpy(out_data, &out_data_array[0], size);
+        uint8_t* out_data = (uint8_t*)malloc(size + 1);
+        memcpy(out_data+1, &out_data_array[0], size);
```

Comment:
Previously the encoding was done in a second step (GetBufferData), but now that we only have the CompressBuffer, we do it all in one go, by adding the header byte.